app/models/custom_field.rb

```diff
@@ -0,0 +1,45 @@
+# == Schema Information
+#
+# Table name: custom_fields
+#
+#  created_at          :datetime         not null
+#  custom_field_set_id :integer
+#  data_type           :string
+#  id                  :integer          not null, primary key
+#  internal_name       :string
+#  label               :string
+#  parent_id           :integer
+#  position            :integer
+#  updated_at          :datetime         not null
+#
+# Indexes
+#
+#  index_custom_fields_on_custom_field_set_id  (custom_field_set_id)
+#
+# Foreign Keys
+#
+#  fk_rails_b30226ad05  (custom_field_set_id => custom_field_sets.id)
+#
+
+class CustomField < ActiveRecord::Base
+  include Translatable
+
+  belongs_to :custom_field_set
+  # note, the custom field form layout can be hierarchially nested
+  belongs_to :parent, class_name: 'CustomField'
```

Comment:
Yeah, that occurred to me also.  The exact requirements around the questionnaire feature isn't completely clear to me yet though, so I didn't want to bite off any more depth into the modelling work here until that was more clear.